src/discussions/common/AuthorLabel.jsx

```diff
@@ -109,13 +110,21 @@ const AuthorLabel = ({
     </>
   ), [author, authorLabelMessage, authorToolTip, icon, isRetiredUser, postCreatedAt, showTextPrimary, alert]);
 
+  const { pathname } = discussionsPath(Routes.LEARNERS.POSTS, {
+    0: enableInContextSidebar ? 'in-context' : undefined,
+    learnerUsername: author,
+    courseId,
+  })();
+
+  const { search } = useLocation();
+
   return showUserNameAsLink
     ? (
       <div className={className}>
         <Link
           data-testid="learner-posts-link"
           id="learner-posts-link"
-          to={generatePath(Routes.LEARNERS.POSTS, { learnerUsername: author, courseId })}
+          to={`${pathname}${enableInContextSidebar ? search : ''}`}
```

Comment:
Post/Comment/Response author’s name should be just simple text in inContext view
[https://2u-internal.atlassian.net/browse/INF-1224](INF-1224)